pages/docs/learn/inngest-functions.mdx

```diff
@@ -24,10 +24,12 @@ export default inngest.createFunction(
   { event: "shop/product.imported" },
   // handler function
   async ({ event, step }) => {
-    // here goes the business logic
-    return copyAllImagesToS3(event.data.imageURLs);
-    // if you need to perform more than one task,
-    // wrap each one in step.run()
+    // Here goes the business logic
+    // By wrapping code in steps, it will be retried automatically on failure
+    const copyImages = await step.run("copy-images-to-s3", async () => {
+      return copyAllImagesToS3(event.data.imageURLs);
+    });
+    // You can include numerous steps in your function
```

Comment:
It might be good to show something using the result of the first step. This might be easier to illustrate the concept of oh this is what goes in a step. Without it, a single step feels a little unclear I think

```suggestion
    const s3Urls = await step.run("copy-images-to-s3", async () => {
      return copyAllImagesToS3(event.data.imageURLs);
    });
    // You can include numerous steps in your function
    await step.run('resize-images', async () => {
      await resizer.bulk({ urls: s3Urls, quality: 0.9, maxWidth: 1024 });
    })
```